plugins/node/opentelemetry-instrumentation-express/src/types.ts

```diff
@@ -70,6 +70,13 @@ export type LayerMetadata = {
 
 export type IgnoreMatcher = string | RegExp | ((name: string) => boolean);
 
+export type SpanNameHook = (
+  request: Request,
+  route: string,
+  layerType: ExpressLayerType | null,
+  defaultPath: string
+) => string;
```

Comment:
I suggest to pass a single `info` object so if we need to extend it in the future, we won't end up with huge funcion signatures with many unused variables.
This pattern is what we use for most instrumentation hooks